### ISSUE TITLE
How can I match a fingerprint against HTML body?

### ISSUE THREAD

**Issue by stoneface919** [rapid7/recog#469]
As the question states. is there any way to match against an HTML body.

Can I use something like this? :

`<fingerprints` matches="html_body" protocol="http" database_type="service" preference="0.90">`

**Comment by mkienow-r7** [comment]
@stoneface919 We do not currently have a database for matching against HTML bodies. Regular expressions are not ideal here. We have plans to explore XPath based solutions in the future.

**Comment by stoneface919** [comment]
I see now thank you for your reply. Do you happen to know if this recog is being used on InsightVM?

**Comment by mkienow-r7** [comment]
Yes, recog is in use in InsightVM. Checkout this older blog post, [Self-Isolation, Home Networking, and Open Source: Recog and Rumble](https://www.rapid7.com/blog/post/2020/04/08/self-isolation-home-networking-and-open-source-recog-and-rumble/), which mentioned this detail and other products using the recog project!

**Comment by jvoisin** [comment]
XPath wouldn't be able to match things like this:

```
<script type="text/javascript">//<![CDATA[
        FbxConf = {};
        FbxConf.apiBaseUrl = '/api/latest/';
        FbxConf.uploadBaseUrl = '/api/latest/upload/';
        FbxConf.websocketBaseUrl =
            (document.location.protocol == "https:" ? "wss://" : "ws://") +
            document.location.host + '/api/latest/ws/';
        FbxConf.csrfToken = '';
        FbxConf.firmwareVersionMajor = '4';
        FbxConf.firmwareVersionMinor = '7';
        // […]
//]]></script>
```

it's from the landing page of a [freebox](https://en.wikipedia.org/wiki/Freebox), leaking its major and minor version.
